src/transformers/generation/utils.py

```diff
@@ -1150,12 +1150,22 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
 
         # Encoder-Decoder models may also need Encoder arguments from `model_kwargs`
         if self.config.is_encoder_decoder:
+            base_model = getattr(self, self.base_model_prefix, None)
+
             # allow encoder kwargs
-            encoder_model_args = set(inspect.signature(self.encoder.forward).parameters)
+            encoder = getattr(self, "encoder", None)
+            if encoder is None:
+                encoder = getattr(base_model, "encoder")
```

Comment:
Assume, in this case, the base model exists and should have `encoder` attribute.
(so far it works well. If something goes wrong in the future, it might suggest the model code has some issue)

Reply:
Hmm this doesn't generalise to MusicGen where we have two different encoders, each of which is named differently to `self.encoder` (a text encoder named `self.text_encoder` and an audio encoder named `self.audio_encoder`).

Do the methods `get_encoder()` and `get_decoder()` not work here?

Reply:
Thanks @sanchit-gandhi 


Good suggestion for `get_encoder` and `get_decoder`, but not 100% to what we need in this PR.

I have a quick fix to not to failing the test, but let's discuss later for a general fix.